docs/neo-n3/dapp_development/transactions.md

```diff
@@ -90,30 +55,25 @@ Account account = Account.fromWIF("L24Qst64zASL2aLEKdJtRLnbnTbqpcRNWkWJ3yhDh2CLU
 byte[] script = new ScriptBuilder().contractCall(NeoToken.SCRIPT_HASH, "symbol", null)
                                    .toArray();
 
-Transaction tx = new TransactionBuilder(neow3j)
+TransactionBuilder builder = new TransactionBuilder(neow3j)
     .script(script)
-    .signers(AccountSigner.calledByEntry(account))
-    .sign();
+    .signers(AccountSigner.calledByEntry(account));
 
-NeoSendRawTransaction response = tx.send();
+// Build the transaction
+Transaction tx = builder.sign()
+                        .send();
 ```
 
-For manually adding witnesses to a transaction continue reading the next section.
+To manually add witnesses to a transaction, continue reading the next section.
 
 ## Signing Transactions
 
-As you have seen in the last section, the `TransactionBuilder` offers a `sign()` method. It attempts to add the correct
-signatures according to what it finds in the list of signers. If you add an `AccountSigner` that holds a private key,
-the `sign()` method can automatically create the signature/witness for that signer. If the account doesn't have a key,
-e.g., because it is a multi-sig account, you will need to provide the witness manually.
-Witnesses for `ContractSigners` are also added automatically with `sign()`.
+As demonstrated in the previous section, the `TransactionBuilder` offers a `sign()` method to add the correct signatures based on the provided signers. If you use an `AccountSigner` with a private key, the `sign()` method will automatically generate the required signature/witness. However, if the account does not have a key (e.g., for a multi-sig account), you'll need to provide the witness manually.
```

Comment:
@thedanielmark:

I generally dislike mentioning "previous section" or "last section" because it becomes more and more challenging to keep this up to date if we make changes/updates to the text over time. It's almost impossible to keep track of "previous" or "last" section text.

My suggestion: if we want to keep the text as is (and it's fine for me!), we need to AT LEAST add a link to the section we're referring to. This way, if we change that section, we can find all occurrences pointing to it.

Just FYI, @csmuller @mialbu 👆